x-pack/plugins/cases/server/services/cases/transform.ts

```diff
@@ -0,0 +1,240 @@
+import { CaseAttributes, CaseFullExternalService, noneConnectorId } from '../../../common';
+import {
+  findConnectorIdReference,
+  transformFieldsToESModel,
+  transformESConnectorOrUseDefault,
+  transformESConnectorToExternalModel,
+} from '../transform';
+
+export function transformUpdateResponsesToExternalModels(
+  response: SavedObjectsBulkUpdateResponse<ESCaseAttributes>
+): SavedObjectsBulkUpdateResponse<CaseAttributes> {
+  return {
+    ...response,
+    saved_objects: response.saved_objects.map((so) => ({
+      ...so,
+      ...transformUpdateResponseToExternalModel(so),
+    })),
+  };
+}
+
+export function transformUpdateResponseToExternalModel(
+  updatedCase: SavedObjectsUpdateResponse<ESCaseAttributes>
+): SavedObjectsUpdateResponse<CaseAttributes> {
+  const { connector, external_service, ...restUpdateAttributes } = updatedCase.attributes ?? {};
+
+  const transformedConnector = transformESConnectorToExternalModel({
+    // if the saved object had an error the attributes field will not exist
+    connector,
+    references: updatedCase.references,
+    referenceName: connectorIdReferenceName,
+  });
+
+  let externalService: CaseFullExternalService | null | undefined;
+
+  // if external_service is not defined then we don't want to include it in the response since it wasn't passed it as an
+  // attribute to update
+  if (external_service !== undefined) {
+    externalService = transformESExternalService(external_service, updatedCase.references);
+  }
+
+  return {
+    ...updatedCase,
+    attributes: {
+      ...restUpdateAttributes,
+      ...(transformedConnector && { connector: transformedConnector }),
+      // if externalService is null that means we intentionally updated it to null within ES so return that as a valid value
+      ...(externalService !== undefined && { external_service: externalService }),
+    },
+  };
+}
+
+export function transformAttributesToESModel(
+  caseAttributes: CaseAttributes
+): {
+  attributes: ESCaseAttributes;
+  references?: CaseSavedObjectReference[];
+};
+export function transformAttributesToESModel(
+  caseAttributes: Partial<CaseAttributes>
+): {
+  attributes: Partial<ESCaseAttributes>;
+  references?: CaseSavedObjectReference[];
+};
+export function transformAttributesToESModel(
+  caseAttributes: Partial<CaseAttributes>
+): {
+  attributes: Partial<ESCaseAttributes>;
+  references?: CaseSavedObjectReference[];
+} {
+  const { connector, external_service, ...restAttributes } = caseAttributes;
+
+  let transformedAttributes: Partial<ESCaseAttributes> = { ...restAttributes };
+  let pushConnectorId: string | undefined | null;
+
+  if (external_service) {
+    let restExternalService: ExternalServicesWithoutConnectorId | null | undefined;
+    ({ connector_id: pushConnectorId, ...restExternalService } = external_service);
+    transformedAttributes = {
+      ...transformedAttributes,
+      external_service: restExternalService,
+    };
+  } else if (external_service === null) {
```

Comment:
I think we should return `external_service: null` for `external_service === undefined`

Reply:
`external_service` will be `undefined` when a patch request is sent and it doesn't include `external_service` in the request. We don't want to set it to `null` in this situation because we'd overwrite whatever is actually in the saved object. For example if a user pushes a case, `external_service` will be populated. Then if the user changes the title of the case, a patch request will be sent that does not include `external_service`. In this situation we don't want to populate the failed for the saved object client to avoid overwriting whatever is already there.

Reply:
Oh yes, you are right! Thanks!